### PR TITLE
Added coverage information to the editor

### PR DESCRIPTION
**Pull request by kendrickcurtis**
coverage.ts is ported from codacyde, my original PoC for an IDE extension. I haven't changed the standard decorations -- light and dark mode are available.

### CODE REVIEW

**Comment by kendrickcurtis** [comment]
What's left to do here -- looks like when you load a PR it doesn't auto trigger the coverage to appear in the current active editor.
Separately, there should probably be a toggle somewhere to turn off showing coverage highlights in case it's interacting badly with another plugin.

File: package.json [hunk 1]
```diff
@@ -242,14 +253,15 @@
...
-    "fetch-api": "curl https://artifacts.codacy.com/api/codacy-api/33.2.15/apiv3.yaml -o ./api-v3/api-swagger.yaml",
+    "fetch-api": "curl https://artifacts.codacy.com/api/codacy-api/33.11.11/apiv3.yaml -o ./api-v3/api-swagger.yaml",
     "generate-api": "rm -rf ./src/api/client && openapi --input ./api-v3/api-swagger.yaml --output ./src/api/client --useUnionTypes --indent 2 --client axios",
     "update-api": "npm run fetch-api && npm run generate-api",
     "postinstall": "npm run update-api",
     "check-types": "tsc --noEmit"
   },
   "devDependencies": {
     "@codacy/openapi-typescript-codegen": "^0.0.2",
+    "@openapitools/openapi-generator-cli": "^2.7.0",
```
Comment (alerizzo) [inline package.json:264]: was this one added for something? is it used? needed?

File: package.json [hunk 2]
```diff
@@ -149,6 +149,12 @@
         "category": "Codacy commands",
         "icon": "$(arrow-right)"
       },
+      {
+        "command": "codacy.pr.toggleCoverageOff",
```
Comment (alerizzo) [inline package.json:153]: is there a `toggleCoverageOn`?

Comment (kendrickcurtis) [reply package.json]: (fixed - this is shrapnel from a "cleverer" solution I wasn't clever enough to implement)

File: package.json (hunk 1 above)
Comment (kendrickcurtis) [reply package.json]: removed - part of my fiddling around getting the updated api version

**Review by alerizzo — DISMISSED** [review]

**Review by alerizzo — APPROVED** [review]

**Pull request merged**